tools/build_defs/repo/jvm.bzl

```diff
@@ -123,25 +125,51 @@ def _jvm_import_external(repository_ctx):
         "",
     ]))
 
-# This method is public for usage in android.bzl macros
-def convert_artifact_coordinate_to_urls(artifact, server_urls, packaging):
-    """This function converts a Maven artifact coordinate into URLs."""
+def _should_fetch_sources_in_current_env(repository_ctx):
+    env_bazel_jvm_fetch_sources = repository_ctx.os.environ.get(_FETCH_SOURCES_ENV_VAR_NAME, "true")
+    return env_bazel_jvm_fetch_sources.lower() == "true"
 
+def _decode_maven_coordinates(artifact, default_packaging):
     parts = artifact.split(":")
-    group_id_part = parts[0].replace(".", "/")
+    group_id = parts[0]
     artifact_id = parts[1]
     version = parts[2]
-    classifier_part = ""
+    classifier = None
+    packaging = default_packaging
     if len(parts) == 4:
         packaging = parts[2]
         version = parts[3]
     elif len(parts) == 5:
         packaging = parts[2]
-        classifier_part = "-" + parts[3]
+        classifier = parts[3]
         version = parts[4]
 
-    final_name = artifact_id + "-" + version + classifier_part + "." + packaging
-    url_suffix = group_id_part + "/" + artifact_id + "/" + version + "/" + final_name
+    return struct(
+        group_id = group_id,
+        artifact_id = artifact_id,
+        version = version,
+        classifier = classifier,
+        packaging = packaging,
+    )
+
+# This method is public for usage in android.bzl macros
+def convert_artifact_coordinate_to_urls(artifact, server_urls, packaging):
+    """This function converts a Maven artifact coordinate into URLs."""
+    coordinates = _decode_maven_coordinates(artifact, packaging)
+    return _convert_coordinates_to_urls(coordinates, server_urls)
+
+def _convert_coordinates_to_urls(coordinates, server_urls):
```

Comment:
AFAIK it’s maven coordinates and not coordinate. I think, not sure, that it’s because this is a multi dimensional system (groupId, artifactId, etc) where groupId is a coordinate on one axis, artifactId is a coordinate on another axis and so forth. Together they are the coordinates of an artifact.

Reply:
Ok, I'll accept the convention :) 

But in that case can you change the above method to be coordinateS for consistency? 

And there is still only one URL, right?

Reply:
1. Ah, you're right. I am inclined to agree with your assumption that it's only used in android.bzl, but it's not a good idea to lump that change into here so let's leave it 
2. I missed the unchanged lines and thought it was returning just a single URL. So everything looks good to me!